beacon-chain/blockchain/service.go

```diff
@@ -458,6 +439,38 @@ func (s *Service) initializeChainInfo(ctx context.Context) error {
 		return errors.Wrap(err, "could not get finalized state from db")
 	}
 
+	if flags.Get().HeadSync {
+		headBlock, err := s.beaconDB.HeadBlock(ctx)
+		if err != nil {
+			return errors.Wrap(err, "could not retrieve head block")
+		}
+		headEpoch := helpers.SlotToEpoch(headBlock.Block.Slot)
+		var epochsSinceFinality uint64
+		if headEpoch > finalized.Epoch {
+			epochsSinceFinality = headEpoch - finalized.Epoch
+		}
+		// Head sync when node is far enough beyond known finalized epoch,
+		// this becomes really useful during long period of non-finality.
+		if epochsSinceFinality >= headSyncMinEpochsAfterCheckpoint {
```

Comment:
In my opinion we should fail out loud here, if a user wants to use `head-sync` and the node can't perform `head-sync` for whatever reason, ignore and just proceed silently is bad

Reply:
It is not that node can't perform head sync, it is just not that necessary - we are **not** in a non-finality period, and finalized checkpoint is nearby, so node as well can restart from there.

If we fail loudly - then, node stops (?) and effectively, we are forcing user to run w/o `--head-sync` flag if there's no non-finality, and then on non-finality rely on that flag i.e. force user to juggle with flags, which is something we should avoid (in original issue raised by user, user was advised to run w/o `--head-sync` when syncing from genesis, to switch to `--head-sync` once entering non-finality -- that was good for a workaround, but for a normal workflow it is not that user-friendly).

Thoughts?